test/unit/slot/distribution.js

```diff
@@ -42,4 +42,59 @@ describe('slot/distribution', function () {
     expect(slot.assignedNodes()[1]).to.equal(lightText);
   });
 
+  describe('distributes to the slot that is owned by the current shadow root, not slots of descendant shadow roots', () => {
+    it('for default slots', () => {
+      const host1 = document.createElement('host1');
+      const host2 = document.createElement('host2');
+      const slot1 = document.createElement('slot');
+      const slot2 = document.createElement('slot');
+      const test = document.createTextNode('test');
+
+      // Host2 must be set up and then added to host1's shadow root so that
```

Comment:
For these more complicated setups, do you think a comment like

```
/*
<host1>
    <test></test>
    <_shadow_root>
        <slot></slot>
        <host2>
            <slot></slot>
            <_shadow_root>
                <slot></slot>
            </_shadow_root>
        </host2>
    </_shadow_root>
</host1>
*/
```

might help?